### DISCUSSION TITLE
Icongenie in Cordova - ERROR: platform not installed!

### DISCUSSION THREAD

**Discussion by vahost** [quasarframework/quasar#13289] [CLI - Capacitor / Cordova modes]
I'm encountering an error in my app with Icongenie in Cordova only. 

Here's my icongenie v printout:

 VERIFYING with the following options:
 ================
 Root folder..... /Users/waltthiessen/Documents/loa-today
 Assets of....... spa | cordova | capacitor | electron
 Assets filter... none
 ================

 Mode SPA 

 SIZE OK - png:          public/icons/favicon-128x128.png 
 SIZE OK - png:          public/icons/favicon-96x96.png 
 SIZE OK - png:          public/icons/favicon-32x32.png 
 SIZE OK - png:          public/icons/favicon-16x16.png 
 SIZE OK - ico:          public/favicon.ico 

 Mode CORDOVA 

 SIZE OK - png:          src-cordova/res/android/ldpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/android/mdpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/android/hdpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/android/xhdpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/android/xxhdpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/android/xxxhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-ldpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-ldpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-mdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-mdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-hdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-hdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-xhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-xhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-xxhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-xxhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-land-xxxhdpi.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/android/splash-port-xxxhdpi.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-20@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-20@3x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-29.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-29@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-29@3x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-40@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-60@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-60@3x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-20.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-20@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-40.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-50.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-50@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-72.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-72@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-76.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-76@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-83.5@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-1024.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-24@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-27.5@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-29@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-29@3x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-40@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-44@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-50@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-86@2x.png ERROR: platform not installed!
 SIZE OK - png:          src-cordova/res/ios/icon-98@2x.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@2x~iphone~anyany.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@2x~iphone~comany.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@2x~iphone~comcom.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@3x~iphone~anyany.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@3x~iphone~anycom.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@3x~iphone~comany.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@2x~ipad~anyany.png ERROR: platform not installed!
 SIZE OK - splashscreen: src-cordova/res/screen/ios/Default@2x~ipad~comany.png ERROR: platform not installed!

 Mode CAPACITOR 

 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-hdpi/ic_launcher_foreground.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-hdpi/ic_launcher.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-mdpi/ic_launcher_foreground.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-mdpi/ic_launcher.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xhdpi/ic_launcher_foreground.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_foreground.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_foreground.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png 
 SIZE OK - png:          src-capacitor/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-land-mdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-port-mdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-land-hdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-port-hdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-land-xhdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-port-xhdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-land-xxhdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-port-xxhdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-land-xxxhdpi/splash.png 
 SIZE OK - splashscreen: src-capacitor/android/app/src/main/res/drawable-port-xxxhdpi/splash.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-20x20@1x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-20x20@2x-1.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-20x20@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-20x20@3x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-29x29@1x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-29x29@2x-1.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-29x29@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-29x29@3x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-40x40@1x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-40x40@2x-1.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-40x40@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-40x40@3x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-60x60@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-60x60@3x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-76x76@1x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-76x76@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-83.5x83.5@2x.png 
 SIZE OK - png:          src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset/AppIcon-512@2x.png 
 SIZE OK - splashscreen: src-capacitor/ios/App/App/Assets.xcassets/Splash.imageset/splash-2732x2732-1.png 
 SIZE OK - splashscreen: src-capacitor/ios/App/App/Assets.xcassets/Splash.imageset/splash-2732x2732-2.png 
 SIZE OK - splashscreen: src-capacitor/ios/App/App/Assets.xcassets/Splash.imageset/splash-2732x2732.png 

 Mode ELECTRON 

 SIZE OK - icns:         src-electron/icons/icon.icns 
 SIZE OK - ico:          src-electron/icons/icon.ico 
 SIZE OK - png:          src-electron/icons/icon.png 

SPA, Capacitor, and Electron all install correctly, but Cordova generates this weird error for each of the png files.

The correct icons actually get generated in src-cordova/res/ios but the error persists.

The result is that my app icon uses the default icongenie icon when running:
quasar dev -m cordova -T ios

Even more strangely, I'm getting this on my Mac, and I was getting it on my PC as well. I did something (not sure what) that cleared the error on my PC, but I can't seem to clear it on my Mac.

I've tried globally reinstalling icongenie.
I've tried rerunning icongenie g -i path/to/file and it shows no errors during the run.

I don't see any other errors.

I don't know what else to do. Help!

**Comment by metalsadman** [comment]
clean your deps and install again. if you've installed the other modes via dev command, try adding the modes manually via `quasar mode add xxx`

**Reply by vahost** [reply]
@metalsadman Here is what I did based upon your recommendation.

Deleted nodes_module folder
Deleted src-cordova
Deleted src-capacitor
Deleted yarn.lock
Removed references to cordova and capacitor from package.json

Then I ran:
yarn install
quasar mode add cordova
quasar mode add capacitor
quasar dev -m cordova -T ios   &nbsp;  &nbsp;  &nbsp; -  it complained that  @capacitor/core isn't installed
npm install --save @capacitor/core
quasar dev -m cordova -T ios   &nbsp;  &nbsp;  &nbsp; -  default icongenie icon showing for app icon
yarn global add @quasar/icongenie
icongenie g -i public/<logo.file>
icongenie v   &nbsp;  &nbsp;  &nbsp; -  I still get the error for all src-cordova files

What am I overlooking?
